cmd/automountServiceAccountToken.go

```diff
@@ -5,53 +5,50 @@ import (
 	"github.com/spf13/cobra"
 )
 
-func checkAutomountServiceAccountToken(result *Result) {
-
-	// Check for use of deprecated service account name
-	if result.dsa != "" {
-		result.err = 1
-	}
-
-	if result.token != nil {
-		// automountServiceAccountToken = true, and serviceAccountName is blank (default: default)
-		if *result.token && result.sa == "" {
-			result.err = 2
-		}
-	} else {
-		// automountServiceAccountToken = nil (default: true), and serviceAccountName is blank (default: default)
-		if result.sa == "" {
-			result.err = 3
-		}
-	}
-}
-
 func printResultASAT(results []Result) {
-
 	for _, result := range results {
-		if result.dsa != "" {
+		switch result.err {
+		case ErrorServiceAccountTokenDeprecated:
 			log.WithFields(log.Fields{
 				"type":               result.kubeType,
 				"namespace":          result.namespace,
 				"name":               result.name,
 				"serviceAccount":     result.dsa,
 				"serviceAccountName": result.sa,
 			}).Warn("deprecated serviceAccount detected (sub for serviceAccountName)")
-		}
-
-		if result.err == 2 {
+		case ErrorServiceAccountTokenNoName:
 			log.WithFields(log.Fields{
 				"type":      result.kubeType,
 				"namespace": result.namespace,
 				"name":      result.name,
 			}).Error("automountServiceAccountToken = true with no serviceAccountName")
-		} else if result.err == 3 {
+		case ErrorServiceAccountTokenNIL:
 			log.WithFields(log.Fields{
 				"type":      result.kubeType,
 				"namespace": result.namespace,
 				"name":      result.name,
 			}).Error("automountServiceAccountToken nil (mounted by default) with no serviceAccountName")
 		}
+	}
+}
+
+func checkAutomountServiceAccountToken(result *Result) {
+	// Check for use of deprecated service account name
+	if result.dsa != "" {
+		result.err = ErrorServiceAccountTokenDeprecated
+		return
+	}
 
+	if result.token != nil && *result.token && result.sa == "" {
+		// automountServiceAccountToken = true, and serviceAccountName is blank (default: default)
+		result.err = ErrorServiceAccountTokenNoName
```

Comment:
Error name does not contain all the information. Need to cover both aspect of `and`